lib/galaxy/tools/evaluation.py

```diff
@@ -543,9 +545,19 @@ def __build_environment_variables(self):
             directory = self.local_working_directory
             environment_variable = environment_variable_def.copy()
             environment_variable_template = environment_variable_def["template"]
+            inject = environment_variable_def.get("inject")
+            if inject == "api_key":
+                if self._user:
+                    from galaxy.managers import api_keys
+                    environment_variable_template = api_keys.ApiKeyManager(self.app).get_or_create_api_key(self._user)
+                else:
+                    environment_variable_template = ""
```

Comment:
Can we raise an Exception here ? They will be displayed in the dataset misc blurb thingy. If the tool asked for `inject` we should probably error with a useful message.

Reply:
Discussed this here https://github.com/galaxyproject/galaxy/pull/8897#discussion_r343050378. I think if the tool doesn't specify ``require_login="true"`` then the API key should be assumed to be optional.

Reply:
makes sense